Upstream: no
Source: https://patchwork.kernel.org/patch/3833241/

From: David Heidelberger <david.heidelberger@ixit.cz>
Date: Fri, 14 Mar 2014 15:23:32 +0000
Subject: [PATCH] [1/4] glibc specific inclusion of sysinfo.h in kernel.h

including sysinfo.h from kernel.h makes no sense whatsoever,
but removing it breaks glibc's userspace header,
which includes kernel.h instead of sysinfo.h from their sys/sysinfo.h.
this seems to be a historical mistake.
on musl, including any header that uses kernel.h directly or indirectly
plus sys/sysinfo.h will produce a compile error due to redefinition of
struct sysinfo.h.
so for now, only include it on glibc in order not to break their 
headers.

Signed-off-by: John Spencer <maillist-linux@barfooze.de>
Signed-off-by: David Heidelberger <david.heidelberger@ixit.cz>
---

--- ./include/uapi/linux/kernel.h.orig
+++ ./include/uapi/linux/kernel.h
@@ -1,7 +1,9 @@
 #ifndef _UAPI_LINUX_KERNEL_H
 #define _UAPI_LINUX_KERNEL_H
 
+#ifdef __GLIBC__
 #include <linux/sysinfo.h>
+#endif
 
 /*
  * 'kernel.h' contains some often-used function prototypes etc

